arch/arm/boot/dts/overlays/fancom-whc-overlay.dts

```diff
@@ -0,0 +1,191 @@
+			};
+
+			gpio_pwr_fail: gpio_pwr_fail {
+				brcm,pins = <2>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+			};
+
+		};
+	};
+
+	fragment@1 { /* Debug port */
+		target = <&uart1_pins>;
+		__overlay__ {
+			brcm,pins = <14 15>; /* pin 51, 55 */
+			brcm,function = <BCM2835_FSEL_ALT0>;
+			brcm,pull = <0 2>;
+		};
+	};
+
+	fragment@2 { /* Debug port */
+		target = <&uart1>;
+		__overlay__ {
+			status = "okay";
+			pinctrl-names = "default";
+			pinctrl-0 = <&uart1_pins>;
+		};
+	};
+
+
+	/* Fixes because RPI does stupid stuff. */
+	fragment@3 {
+		target = <&phy1>;
+		__overlay__ {
+			reg = <0>;
+		};
+	};
+
+	fragment@4 {
+		target = <&i2c0>;
+		__overlay__ {
+			#gpio-cells = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "okay";
+
+			pinctrl-names = "default";
+			pinctrl-0 = <&i2c0_pins>;
+			clock-frequency = <400000>;
+
+			pcf8523@68 {
+				compatible = "nxp,pcf8523";
+				reg = <0x68>;
+				quartz-load-femtofarads = <7000>; /* in femto Farad (fF) */
+
+				//interrupt = <SPI_GIC 53 IRQ_TYPE_LEVEL_HIGH>; // gpio6, int = "or of all i2c".
+			};
+		};
+	};
+
+	fragment@5 {
+		target = <&pcie0>; // PCIE2?
+		__overlay__ {
+			pci@2,0 {
+				// PCIe NVMe SSD M.2 slot, but how?
+
+				/*
+				 * https://en.wikipedia.org/wiki/M.2
+				 * https://forums.developer.nvidia.com/t/installing-nvme-drive/49957
+				 * https://community.nxp.com/t5/i-MX-Processors/IMX7D-PCIE-Support/td-p/810018
+				 */
+			};
+		};
+	};
+
+	fragment@6 {
+		target-path = "/"; // = <&gpio>;
+		__overlay__ {
+			leds_rgb {
+				compatible = "gpio-leds"; /* See leds-gpio.yaml */
+				led_red {
+					gpios = <&gpio 24 GPIO_ACTIVE_LOW>; /* pin 45 */
+					color = <LED_COLOR_ID_RED>;
+					default-state = "off";
+				};
+
+
+				led_green {
+					gpios = <&gpio 22 GPIO_ACTIVE_LOW>; /* pin 46 */
+					color = <LED_COLOR_ID_GREEN>;
+					default-state = "off";
+				};
+
+				led_blue {
+					gpios = <&gpio 23 GPIO_ACTIVE_LOW>; /* pin 47 */
+					color = <LED_COLOR_ID_BLUE>;
+					default-state = "off";
+				};
+			};
+
+			gpio-keys {
+				compatible = "gpio-keys";
+				autorepeat; //The autorepeat property allows holding the key to continuously generate events.
+				pinctrl-names = "default";
+
+				pinctrl-0 = <&gpio_restore>;
+				pinctrl-1 = <&gpio_pewake>;
+				pinctrl-2 = <&gpio_rtc_int>;
+				pinctrl-3 = <&gpio_lan_rst>;
+				pinctrl-4 = <&gpio_pwr_fail>;
+
+				restore {
+					label = "restore";
+					gpios = <&gpio 7 GPIO_ACTIVE_HIGH>; //input
+					linux,code = <0x290>; // 0x290 KEY_MACRO1
+				};
+
+				pewake {
+					label = "nvm_wakeup";
+					gpios = <&gpio 8 GPIO_ACTIVE_HIGH>; //output
+					linux,code = <0x291>; // 0x291 KEY_MACRO2
+				};
+
+				rtc_int {
+					label = "rtc_interrupt";
+					gpios = <&gpio 6 GPIO_ACTIVE_HIGH>; //input
+					linux,code =  <0x292>; // 0x292 KEY_MACRO3
+				};
+
+				lan_rst {
+					label = "lan_reset";
+					gpios = <&gpio 4 GPIO_ACTIVE_HIGH>; //output
+					linux,code = <0x293>; // 0x293 KEY_MACRO4
```

Comment:
No need for a key code, this is just a digital state

Reply:
https://www.kernel.org/doc/Documentation/devicetree/bindings/input/gpio-keys.txt